cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/datastore/DatastoreManager.java

```diff
@@ -333,7 +333,7 @@ private Datastore createDatastore(String dbName, KeyProvider provider) throws Da
             // if it does not exist
 
             //Pass database directory, database name, and SQLCipher key provider
-            DatastoreImpl ds = new DatastoreImpl(dbDirectory, dbName, provider);
+            QueryableDatastore ds = new QueryableDatastore(dbDirectory, dbName, provider);
```

Comment:
@rhyshort cool glad we're in agreement. I do also concede your point that there will be some ugly coupling between (for example) the basic CRUD datastore and the Index/Query 'manager', but that's no worse than what we have at the moment except we can at least hide the complexity behind a façade.